### PR TITLE
[SwiftPM] Add Swift package manager support.

### PR DESCRIPTION
**Pull request by ddunbar**
- This allows llbuild to be used via the Swift package manager.

 - There are a mix of reasonable changes to the source to better work with SwiftPM, and workarounds.

 - The reasonable changes:
    * Force gtest config via source code rather than build system. This actually cleans up the other configs.

 - The workarounds:
    * Unfortunate, but acceptable: we have to rename a file to main.cpp to meet SwiftPM requirements (SR-6432).
    * Add a bunch of symlinks to make include paths work right. This is acceptable, but gross. Potentially will be solved with SwiftPM build settings.

## TODO

* [ ] Add support for running tests.
* [ ] Add better way to handle linking to SQLite & ncurses, so it doesn't require spelling on the command line.
* [x] Add documentation and examples.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Oohhh mai gawd😱😱😱😱😱😱

**Comment by ddunbar** [comment]
Added support for building unit tests:
* This required a couple cleanups, and some workarounds mostly for making the expected way include paths are used in the source code work (e.g., a self-referential src/ symlink in the tree).

Changed to just expose the libllbuild product, and fixed name of `llbuild` tool.

Added support for the swift-build-tool product.

This now is building basically all of the code. The most significant limitation is that there is no way to run the unit tests.

**Comment by ddunbar** [comment]
* Added Swift bindings to SwiftPM build.
* Added a new set of Swift bindings for the BuildSystem library.

**Comment by ddunbar** [comment]
Added an example of computing Conway's Game of Life, using the bindings and the Swift [HTTP server](https://github.com/swift-server/http).

File: examples/GameOfLife/Package.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// swift-tools-version:4.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "GameOfLife",
+    dependencies: [
+        .package(url: "https://github.com/swift-server/http", from: "0.0.0"),
+        .package(url: "../../../llbuild", .branch("WIP-SwiftPM-support")),
```
Comment (aciidb0mb3r) [inline examples/GameOfLife/Package.swift]: Is the branch still valid?

Comment (ddunbar) [reply examples/GameOfLife/Package.swift]: Nope, just pushed a fix for this.

**Comment by ddunbar** [comment]
Updated this to fix rename of ConvertUTF.cpp, which is fixed in SwiftPM now. We still have several workarounds in place, but none of them are invasive enough to prevent landing this.

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
Did some additional testing on Linux, which exposed a few minor issues.

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
@swift-ci, I'd like to talk to you. I really want this change to be tested!!! I know you are busy, but it would just mean so much to me if I could land it tonight.

I know you will feel better in the morning, but maybe, just maybe, you could do this last one before you sleep?

@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Pull request merged**